### ISSUE TITLE
Avoid world change

### ISSUE THREAD

**Issue by UltraBlackLinux** [cabaletta/baritone#1792]
## Describe your suggestion With as much detail as possible, describe what your suggestion would do for Baritone.
So it basically pathes but tries to avoid breaking/Placing as long as it wouldnt take to much time

## Settings If applicable, what settings/customizability should be offered to tweak the functionality of your suggestion.
 
#worldchangetimeout int (In seconds or mlliseconds. This stops baritone from avoiding to place/break blocks if the calculated time is higher than this value)

#avoidworldchange boolean

## Context Describe how your suggestion would improve Baritone, or the reason behind it being added. 

This wouls be useful for pathing in a base and not wanting baritone to break holes into the walls.

## Final checklist
 - [x] I know how to properly use check boxes
 - [ ] I have not used any OwO's or UwU's in this issue.

**Comment by dominikaaaa** [comment]
Completely disallowing breaking / placing would make it impossible to path in some cases, which is why baritone only tries to avoid doing so.

**Comment by dominikaaaa** [comment]
Iirc, as long as there is a valid path with `allowBreak` disabled, it won't break stuff

**Comment by UltraBlackLinux** [comment]
but its only supposed to *avoid* breaking. if youre in an enclosed space you propably going to wanna do this since you want to get out but with the least damage to the world.

**Comment by dominikaaaa** [comment]
Yes, and as I said, it'll avoid breaking unless it's impossible not to

**Comment by ZacSharp** [comment]
with allowBreak and allowPlace being false it should never break or place blocks
to avoid those you should set blockBreakAdditionalPenalty and blockPlacementPenalty to higher values.
(already slightly prefers not changing blocks by default)

**Comment by Wahubah** [comment]
I don't think we should change the world too, im fine with how everything is atm
